testing/endtoend/evaluators/fork.go

```diff
@@ -17,15 +17,36 @@ import (
 )
 
 var streamDeadline = 1 * time.Minute
-var startingFork = version.Phase0
+
+// genesisFork returns the fork version at genesis based on the current config.
+// It checks from highest fork to lowest to find which fork is active at epoch 0.
+func genesisFork() int {
+	cfg := params.BeaconConfig()
+	if cfg.ElectraForkEpoch == 0 {
+		return version.Electra
+	}
+	if cfg.DenebForkEpoch == 0 {
+		return version.Deneb
+	}
+	if cfg.CapellaForkEpoch == 0 {
+		return version.Capella
+	}
+	if cfg.BellatrixForkEpoch == 0 {
+		return version.Bellatrix
+	}
+	if cfg.AltairForkEpoch == 0 {
+		return version.Altair
+	}
+	return version.Phase0
+}
 
 // AltairForkTransition ensures that the Altair hard fork has occurred successfully.
 var AltairForkTransition = types.Evaluator{
 	Name: "altair_fork_transition_%d",
 	Policy: func(e primitives.Epoch) bool {
 		altair := policies.OnEpoch(params.BeaconConfig().AltairForkEpoch)
-		// TODO (11750): modify policies to take an end to end config
-		if startingFork == version.Phase0 {
+		// Only run if we started before Altair
+		if genesisFork() < version.Altair {
 			return altair(e)
 		}
 		return false
```

Comment:
Can you reorganize this code to match other forks? It's easier to reason about these transitions when all of them follow the same pattern.

Reply:
https://github.com/OffchainLabs/prysm/pull/16048/commits/b1a95ebc5edea704f40d21c3de743f09d064902c resolved here